client/src/workspace/layouts/AppsViewLayout.tsx

```diff
@@ -18,17 +19,33 @@ export const AppsViewLayout: React.FC = () => {
           {parse(htmlApp.html as string)}
         </div>
       ) : (
-        <Suspense
-          fallback={
-            <Layout>
-              <Spinner />
-            </Layout>
+        <ErrorBoundary
```

Comment:
This works 👍 

Curious if you have thoughts on
- Would `errorElement` also work here?
https://reactrouter.com/en/main/route/error-element
- Or error handling within the router? https://tanstack.com/query/latest/docs/framework/react/guides/query-functions#handling-and-throwing-errors

Reply:
1) error-element within router is close to this PR's solution. It provide more protection to overall component instead of just the <Suspense> usage.
I tested it by making changing in workspaceRouter like this

```
{
    id: 'app',
    path: ':appId',
    element: <AppsViewLayout />,
    errorElement: <ErrorBoundary />
},
```
We could put it in the workspaceRouter as a single boundary, which will catch everything.
```
<ErrorBoundary>
    {workspaceRouter}
  </ErrorBoundary>
```

The only change would be extra handling for component specific errors. We are not doing it right now, we are simply showing the error, so doing this might help us with catching all errors.

2) QueryFn error handling - even with error handling here, with suspense usage, we will still need ErrorBoundary. This is because there is no visibility into the error like `isError` 


What do you think? Do you want to move the ErrorBoundary one or two levels up to catch everything?

Reply:
Either way. This is good enough for me for now. Perhaps we can bring the error boundary up later if need be